zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -504,60 +512,97 @@ impl<F: Field> SignVerifyChip<F> {
+                ctx,
+                &pk_hash_lo_cell_bytes
+                    .iter()
+                    .zip(&powers_of_256)
+                    .map(|(cell, coeff)| maingate::Term::Assigned(cell, *coeff))
+                    .collect_vec(),
+                F::ZERO,
+                |_, _| Ok(()),
+            )?;
+            let (address_cell_hi, _) = main_gate.decompose(
+                ctx,
+                &pk_hash_hi_cell_bytes
+                    .iter()
+                    .take(N_BYTES_ACCOUNT_ADDRESS - 16)
+                    .zip(&powers_of_256)
+                    .map(|(cell, coeff)| maingate::Term::Assigned(cell, *coeff))
+                    .collect_vec(),
+                F::ZERO,
+                |_, _| Ok(()),
+            )?;
 
             (
-                address,
-                pk_hash_lo
-                    .into_iter()
-                    .zip(pk_hash[12..].iter())
-                    .map(|(assigned, byte)| Term::assigned(assigned.cell(), *byte))
-                    .collect_vec(),
+                Word::new([address_cell_lo, address_cell_hi]),
+                Word::new([pk_hash_cell_lo, pk_hash_cell_hi]),
             )
         };
-        let is_address_zero = main_gate.is_zero(ctx, &address)?;
+
+        let iz_zero_hi = main_gate.is_zero(ctx, &address_cells.hi())?;
+        let iz_zero_lo = main_gate.is_zero(ctx, &address_cells.lo())?;
+        let is_address_zero = main_gate.and(ctx, &iz_zero_lo, &iz_zero_hi)?;
 
         // Ref. spec SignVerifyChip 3. Verify that the signed message in the ecdsa_chip
-        // with RLC encoding corresponds to msg_hash_rlc
-        let msg_hash_rlc = {
-            let zero = main_gate.assign_constant(ctx, F::ZERO)?;
-            let assigned_msg_hash_le = assigned_ecdsa
-                .msg_hash_le
-                .iter()
-                .map(|byte| main_gate.select(ctx, &zero, byte, &is_address_zero))
-                .collect::<Result<Vec<_>, _>>()?;
-            let msg_hash_le = (!padding)
-                .then(|| sign_data.msg_hash.to_bytes())
-                .unwrap_or_default()
-                .map(|byte| Value::known(F::from(byte as u64)));
-            self.assign_rlc_le(
-                config,
+        // corresponds to msg_hash
+        let msg_hash_cells = {
+            let msg_hash_lo_cell_bytes = &assigned_ecdsa.msg_hash_le[..16];
+            let msg_hash_hi_cell_bytes = &assigned_ecdsa.msg_hash_le[16..];
+            let (msg_hash_cell_lo, _) = main_gate.decompose(
                 ctx,
-                chips,
-                "msg_hash",
-                config.q_rlc_evm_word,
-                challenges.evm_word(),
-                assigned_msg_hash_le
+                &msg_hash_lo_cell_bytes
                     .iter()
-                    .zip(msg_hash_le)
-                    .map(|(assigned, byte)| Term::assigned(assigned.cell(), byte)),
-            )?
+                    .zip(&powers_of_256)
```

Comment:
here `zip_eq` is better than `zip`

Reply:
fixed in [24c67ee](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1418/commits/24c67eea3617aaa417a24fd5e1a0f21895fe06f7)